kernel/kernel-api/src/main/java/io/delta/kernel/TableIdentifier.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel;
+
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import io.delta.kernel.annotation.Evolving;
+
+/**
+ * Identifier for a table. e.g. $catalog / $schema / $tableName
```

Comment:
Only table has a `Name` suffix. It would be nice to use the same naming rule in all objects.

Reply:
would you prefer $catalogName etc or $table ? i.e. pro or against Name suffix?